src/js/crud.js

```diff
@@ -0,0 +1,136 @@
+// This class creates a Todo application.
+class TodoApp {
+  //Constructs the Todo App, fetch tasks from local storage and set up event listeners.
+  constructor() {
+    this.inputField = document.getElementById('input');
+    this.returnButton = document.getElementById('return-btn');
+    this.todoTask = JSON.parse(localStorage.getItem('tasks')) || [];
+    this.todoList = document.getElementById('to-do-list');
+    this.render();
+    this.AddTaskPressEnter();
+    this.AddTaskPressButton();
+    this.DeleteTaskButton();
+  }
+  // Sets an event listener for deleting a task when a specific element is clicked
+  DeleteTaskButton() {
+    this.todoList.addEventListener('mousedown', (e) => {
+      if (e.target.closest('.trash-icon')) {
+        this.deleteTask(e.target.dataset.task);
+      }
+    });
+  }
+  // Sets an event listener for adding a task when return button is clicked
+  AddTaskPressButton() {
+    this.returnButton.addEventListener('click', (e) => {
+        this.addTask(this.inputField.value);
+        this.inputField.value = '';
+        e.preventDefault();
+    });
+  }
+  // Sets an event listener for adding task when enter button is pressed
+  AddTaskPressEnter() {
+    this.inputField.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        this.addTask(this.inputField.value);
+        this.inputField.value = '';
+      } 
+    });
+  }
+  //Sorts the tasks based on their index
+  sortTasks() {
+    this.todoTask.sort ((a, b) => a.index - b.index)
+  }
+  // Refreshes the tasks presented in the UI
+  render() {
+    this.sortTasks();
+    this.todoList.innerHTML = '';
+    
+    this.todoTask.forEach((task, i) => {
+      let taskElement = document.createElement('li');
+      taskElement.classList.add('task-container');
+  
+      let taskContent = document.createElement('div');
+      taskContent.classList.add('task');
+      
+      let checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
```

Comment:
- [ ] Please from the project requirement, all changes to the To-Do List should be saved in local storage so make sure that if a status of a task is change for instance from completed `false` to `true` (by checking the checkbox ) the localStorage is updated. As you can see in the video below if I check the checkbox of a task. the completed is not changing from `false` to `true` :smile: 

[Screencast from 07-14-2023 11:34:34 PM.webm](https://github.com/Rebelzob/To-do-app/assets/73854999/080d2291-a945-4028-84de-592ab940249f)

Reply:
Hi Liz this implementation is meant for the next part of the project, not this one.

![image](https://github.com/Rebelzob/To-do-app/assets/52711805/2ab65c76-0094-40a7-a68b-7c298bc121c9)

https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md